store/mapper/codes_test.go

```diff
@@ -0,0 +1,170 @@
+package mapper
+
+import (
+	"fmt"
+	dpbolt "github.com/ONSdigital/dp-bolt/bolt"
+	"github.com/ONSdigital/dp-code-list-api/models"
+	"github.com/johnnadratowski/golang-neo4j-bolt-driver/structures/graph"
+	"github.com/pkg/errors"
+	. "github.com/smartystreets/goconvey/convey"
+	"strconv"
+	"testing"
+)
+
+var (
+	testCodeListID        = "666" // the number of the best \m/
+	testEdition           = "2018"
+	testCode              = "99"
+	errTest               = errors.New("error happened yo")
+	testNodeIdentity      = int64(666) // the number of the best \m/
+	testNodeValue         = "node value"
+	testRelationshipLabel = "relationship label"
+)
+
+func TestCodeResultExtractor_Success(t *testing.T) {
+	Convey("given a valid result", t, func() {
+		node := graph.Node{
+			NodeIdentity: testNodeIdentity,
+			Properties:   map[string]interface{}{"value": testNodeValue},
+		}
+
+		rel := graph.Relationship{
+			Properties: map[string]interface{}{"label": testRelationshipLabel},
+		}
+
+		actual := &models.Code{}
+		extractor := Code(actual, testCodeListID, testEdition)
+
+		Convey("when extractor is called", func() {
+			err := extractor(
+				&dpbolt.Result{
+					Data:  []interface{}{node, rel},
+					Meta:  nil,
+					Index: 0,
+				})
+
+			Convey("then no error is returned and code is populated with the expected values", func() {
+				So(err, ShouldBeNil)
+
+				expected := &models.Code{
+					ID:    strconv.FormatInt(testNodeIdentity, 10),
+					Code:  testNodeValue,
+					Label: testRelationshipLabel,
+					Links: models.CodeLinks{
+						Self: models.Link{
+							Href: fmt.Sprintf("/code-lists/%s/editions/%s/codes/%s", testCodeListID, testEdition, testNodeValue),
+						},
+						Datasets: models.Link{
+							Href: fmt.Sprintf("/code-lists/%s/editions/%s/codes/%s/datasets", testCodeListID, testEdition, testNodeValue),
+						},
+						CodeList: models.Link{
+							Href: fmt.Sprintf("/code-lists/%s", testCodeListID),
+						},
+					},
+				}
+
+				So(expected, ShouldResemble, actual)
+			})
+		})
+	})
+}
+
+func TestCodeResultExtractor_BadTypes(t *testing.T) {
+	node := graph.Node{
+		Properties: map[string]interface{}{
+			"value": testCode,
+		},
+	}
+
+	Convey("given data.0 is not type graph.Node", t, func() {
+		actual := &models.Code{}
+		extractor := Code(actual, testCodeListID, testEdition)
+
+		Convey("when extractor is called", func() {
+			err := extractor(
+				&dpbolt.Result{
+					Data:  []interface{}{"not graph.Node"},
+					Meta:  nil,
+					Index: 0,
+				})
+
+			Convey("then expected error is returned", func() {
+				So(err, ShouldNotBeNil)
+				So(err.Error(), ShouldEqual, "row.Data[0] incorrect type - expected \"graph.Node\", actual \"string\"")
+				So(actual.ID, ShouldBeBlank)
```

Comment:
Instead of these four assertions think about doing just the one as actual should be an empty `Code` object, like so:
`So(actual, ShouldResemble, &models.Code{})`

This also can be applied elsewhere in test.